src/v2/services/implementations/PolkadotWalletService.ts

```diff
@@ -90,6 +92,96 @@ export class PolkadotWalletService extends WalletService implements IWalletServi
     return result;
   }
 
+  // from src/hooks/helper/wallet.ts
+  // @TODO: Need to refactor
+  public async signAndSendWithCustomSignature({
```

Comment:
There is already a very similar signAndSend method in the service. Is it possible to reuse this method here, or extend singAndSend with isCustomSignature?

Reply:
Yes, that makes sense. as overall, I'd like to ask that do we need to keep [useClaim](https://github.com/AstarNetwork/astar-apps/blob/e3103d8d58e20ac9b29e6e6dd09f3d4ba803e20b/src/hooks/useClaimAll.ts#L27) hook since it has vuex store or other hooks depends on?  otherwise, can we retrieve these all information from v2 architecture whether it already implemented or not?

Reply:
In my opinions hooks are fine if they use v2 services